include/GoogleSync/GoogleSyncBase.php

```diff
@@ -0,0 +1,1049 @@
+ * 02110-1301 USA.
+ *
+ * You can contact SugarCRM, Inc. headquarters at 10050 North Wolfe Road,
+ * SW2-130, Cupertino, CA 95014, USA. or at email address contact@sugarcrm.com.
+ *
+ * The interactive user interfaces in modified source and object code versions
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
+
+require_once __DIR__ . '/../../modules/Users/User.php';
+require_once __DIR__ . '/../../modules/Meetings/Meeting.php';
+require_once __DIR__ . '/GoogleSyncExceptions.php';
+
+use SuiteCRM\Utility\SuiteValidator;
+
+/**
+ * Implements Google Calendar Syncing
+ *
+ * @license https://raw.githubusercontent.com/salesagility/SuiteCRM/master/LICENSE.txt
+ * GNU Affero General Public License version 3
+ * @author Benjamin Long <ben@offsite.guru>
+ */
+
+class GoogleSyncBase
+{    
+    /** @var User The SuiteCRM User Bean we're currently working with */
+    protected $workingUser;
+
+    /** @var \Google_Client The Google client object for the current sync job */
+    protected $gClient;
+
+    /** @var \Google_Service_Calendar The Google Calendar Service Object */
+    protected $gService;
+
+    /** @var array The Google AuthcConfig json */
+    protected $authJson = array();
+
+    /** @var string The local timezone */
+    protected $timezone;
+
+    /** @var string The Calendar ID */
+    protected $calendarId;
+
+    /** @var array An array of SuiteCRM meeting id's that we've already synced this session */
+    protected $syncedList = array();
+
+    /** @var object A Database Instance */
+    protected $db;
+
+    /** @var object A Logger Instance */
+    protected $logger;
+
+    /** @var string The log level before we begin */
+    protected $oldLogLevel;
+
+    /**
+     * Class Constructor
+     * 
+     * @param array $sugarConfig - using $sugar_config as a dependency
+     */
+    public function __construct($sugarConfig)
+    {
+        // This sets the log level to a variable that can be set on the command line while running cron.php on the server. It's for debugging only.
+        // EXAMPLE: $ GSYNC_LOGLEVEL=debug php cron.php
+        $this->logger = LoggerManager::getLogger();
+        if (isset($_SERVER['GSYNC_LOGLEVEL'])) { // TODO: is it possible to using $sugar_config instead superglobals? (a global usage is one step closer to clean code than using a superglobal)
```

Comment:
@gymad 
How about I add a line to config_override.php for this. That's almost as easy for the user. It's only ever used for debugging.

Reply:
Done